common/lib/xmodule/xmodule/modulestore/__init__.py

```diff
@@ -276,27 +277,49 @@ def _is_in_bulk_operation(self, course_key, ignore_case=False):
         return self._get_bulk_ops_record(course_key, ignore_case).active
 
 
-class ModuleStoreAssetInterface(object):
+class SortedAssetList(SortedListWithKey):
     """
-    The methods for accessing assets and their metadata
+    List of assets that is sorted based on an asset attribute.
     """
-    @contract(asset_list='SortedListWithKey', asset_id='AssetKey')
-    def _find_asset_in_list(self, asset_list, asset_id):
+    def __init__(self, **kwargs):
+        super(SortedAssetList, self).__init__(**kwargs)
+
+    @contract(asset_id=AssetKey)
+    def find(self, asset_id):
         """
-        Given a asset list that's a SortedListWithKey, find the index of a particular asset.
+        Find the index of a particular asset in the list.
         Returns: Index of asset, if found. None if not found.
         """
         # See if this asset already exists by checking the external_filename.
         # Studio doesn't currently support using multiple course assets with the same filename.
         # So use the filename as the unique identifier.
         idx = None
-        idx_left = asset_list.bisect_left({'filename': asset_id.path})
-        idx_right = asset_list.bisect_right({'filename': asset_id.path})
+        idx_left = self.bisect_left({'filename': asset_id.path})
+        idx_right = self.bisect_right({'filename': asset_id.path})
         if idx_left != idx_right:
             # Asset was found in the list.
             idx = idx_left
         return idx
```

Comment:
I guess maybe because the actual documents in the list contain more that the filename?